superbench/benchmarks/model_benchmarks/pytorch_cnn.py

```diff
@@ -0,0 +1,156 @@
+            return False
+
+        self._target = torch.LongTensor(self._args.batch_size).random_(self._args.num_classes)
+        if self._gpu_available:
+            self._target = self._target.cuda()
+
+        return True
+
+    def _train_step(self, precision):
+        """Define the training process.
+
+        Args:
+            precision (Precision): precision of model and input data, such as float32, float16.
+
+        Return:
+            The step-time list of every training step.
+        """
+        duration = []
+        curr_step = 0
+        while True:
+            for idx, sample in enumerate(self._dataloader):
+                start = time.time()
+                sample = sample.to(dtype=getattr(torch, precision.value))
+                if self._gpu_available:
+                    sample = sample.cuda()
+                self._optimizer.zero_grad()
+                output = self._model(sample)
+                loss = self._loss_fn(output, self._target)
+                loss.backward()
+                self._optimizer.step()
+                end = time.time()
+                curr_step += 1
+                if curr_step > self._args.num_warmup:
+                    # Save the step time of every training/inference step, unit is millisecond.
+                    duration.append((end - start) * 1000)
+                if self._is_finished(curr_step, end):
+                    return duration
+
+    def _inference_step(self, precision):
+        """Define the inference process.
+
+        Args:
+            precision (Precision): precision of model and input data,
+              such as float32, float16.
+
+        Return:
+            The latency list of every inference operation.
+        """
+        duration = []
+        curr_step = 0
+        with torch.no_grad():
+            self._model.eval()
+            while True:
+                for idx, sample in enumerate(self._dataloader):
+                    start = time.time()
+                    sample = sample.to(dtype=getattr(torch, precision.value))
+                    if self._gpu_available:
+                        sample = sample.cuda()
+                    self._model(sample)
+                    if self._gpu_available:
+                        torch.cuda.synchronize()
+                    end = time.time()
+                    curr_step += 1
+                    if curr_step > self._args.num_warmup:
+                        # Save the step time of every training/inference step, unit is millisecond.
+                        duration.append((end - start) * 1000)
+                    if self._is_finished(curr_step, end):
+                        return duration
+
+
+# Register CNN benchmarks.
+# Reference: https://pytorch.org/vision/stable/models.html
+#            https://github.com/pytorch/vision/tree/master/torchvision/models
+MODELS = [
+    'alexnet', 'densenet121', 'densenet169', 'densenet201', 'densenet161', 'googlenet', 'inception_v3', 'mnasnet0_5',
```

Comment:
get it from `torchvision.models.__dict__` dynamically?

Reply:
There have some other terms in the torchvision.models.__dict__, like:

> 'video',  'quantization', 'detection', 'segmentation', 'utils'.

And I don't want to use the v0.1's style, 

> models.mnasnet.__all__[1:]

It's a little trick since the list may be changed.
I should we should claim what models superbench can support explicitly.

Reply:
then do you need to check `hasattr` before register? since different `torchvision` versions may have different lists of models.

Reply:
Sure.